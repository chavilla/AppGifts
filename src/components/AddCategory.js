import React, { useState } from 'react';

import PropTypes from 'prop-types';

const AddCategory = ({setCategory}) => {

    const [inputValue,setInputValue]=useState('');

    const handleChange=(value)=>{
        //console.log('handleChange called');
        setInputValue(value);
    }


    const handleSubmit=(e)=>{
        e.preventDefault();

       // console.log('handle submit');

        if(inputValue.trim().length<3) return false;

        setCategory(state=>[inputValue]);
        setInputValue('');
    }

    return ( 
        <form onSubmit={handleSubmit}>
            <p>{inputValue}</p>
            <input
                type='text'
                onChange={(e)=>handleChange(e.target.value)}
                value={inputValue}
                placeholder='Agrega un HÃ©roe'
            />
        </form>
     );
}

AddCategory.propTypes={
    setCategory:PropTypes.func.isRequired
}

export default AddCategory;