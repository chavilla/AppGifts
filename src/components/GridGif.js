import React, {  } from "react";
import { useFetchGifs } from "../hooks/useFetchGifs";
import GifGridItem from './GifGridItem';
import PropTypes from 'prop-types';

export const GridGif = ({ category }) => {

  const {data, loading}=useFetchGifs(category);


  // console.log(data);

  return (
    <>
      <h2>{category}</h2>
        { loading && <p>Loading</p> }
       <div className="card-grid">
          {data.map((image) => (
            <GifGridItem key={image.id} {...image} />
          ))}
      </div> 
    </>
  );
};

GridGif.propTypes={
  category: PropTypes.string.isRequired
}