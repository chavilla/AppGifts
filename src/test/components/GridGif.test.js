import React from 'react';
import { shallow } from 'enzyme';
import {GridGif} from '../../components/GridGif';
import { useFetchGifs } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs');

describe('GridGrif Component', () => {

   
    
    test('snapshot of the component',()=>{

        const gifts=[{
            id: 'ABC',
            title:'Hola Jesus',
            url:'https://www.chaviweb.com'
        }];

        useFetchGifs.mockReturnValue({
            data: gifts,
            loading:false
        });

        const wrapper=shallow(<GridGif category={'Soy una categoría'}/>)

        expect(wrapper).toMatchSnapshot();
    });

    test('show items with useFetchGifs',()=>{

        const gifts=[{
            id: 'ABC',
            title:'Hola Jesus',
            url:'https://www.chaviweb.com'
        }];

        useFetchGifs.mockReturnValue({
            data: gifts,
            loading:false
        });

        const wrapper=shallow(<GridGif category={'Soy una categoría'}/>)

        expect(wrapper).toMatchSnapshot();


    });
})