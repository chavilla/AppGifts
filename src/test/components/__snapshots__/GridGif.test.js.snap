// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GridGrif Component show items with useFetchGifs 1`] = `
<Fragment>
  <h2>
    Soy una categoría
  </h2>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="ABC"
      key="ABC"
      title="Hola Jesus"
      url="https://www.chaviweb.com"
    />
  </div>
</Fragment>
`;

exports[`GridGrif Component snapshot of the component 1`] = `
<Fragment>
  <h2>
    Soy una categoría
  </h2>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="ABC"
      key="ABC"
      title="Hola Jesus"
      url="https://www.chaviweb.com"
    />
  </div>
</Fragment>
`;
